src/components/token/Tokens.js

```diff
@@ -24,18 +27,101 @@ class Tokens extends React.Component {
             tokens: [],
             hasAfter: false,
             hasBefore: false,
-            searchText: ""
+            searchText: "",
+            sortBy: "uid",
+            order: "asc",
+            page: 1,
+            //This variable stores the pages and its searchAfter params that is passed to explorer-service
+            pageSearchAfter: [],
+            loading: true
         }
     }
 
-    onSearchButtonClicked = () => {
-        // TODO: Implement behavior when search button is clicked
+    componentDidMount = async() => {
+        //"Click" on search to make the first query
+        await this.onSearchButtonClicked();
+        this.setState({loading: false})
+    }
+
+    resetPageSearchAfter = () => {
+        this.setState({pageSearchAfter: [
+            {
+                page: 1,
+                searchAfter: []
+            }
+        ]})
+    }
+
+    getTokens = async (searchAfter) => {
+        const tokensRequest = await tokensApi.getList(this.state.searchText, this.state.sortBy, this.state.order, searchAfter)
+        let tokens = get(tokensRequest, 'data', {hits:[], 'has_next': false})
+        forEach(tokens.hits, (token) => {
+            if(isNil(token.uid)) {
+                token.uid = token.id
+            }
+            token.type = token.nft ? 'NFT' : 'Custom Token'
+        })
+
+        return tokens;
+    }
+
+    onSearchButtonClicked = async () => {
+        //When search button is clicked, results return to the first page
+        this.resetPageSearchAfter();
+        const tokens = await this.getTokens([]);
+
+        this.setState({page: 1, tokens: tokens.hits, hasAfter: tokens.has_next, hasBefore: false})
     }
 
     onSearchTextChanged = (event) => {
         this.setState({ searchText: event.target.value })
     }
 
+    onSearchTextKeyPressed = (event) => {
+        if(event.charCode === 13) {
+            this.onSearchButtonClicked();
+        }
+    }
+
+    nextPageClicked = async (event) => {
+        const nextPage = this.state.page+1
+        let searchAfter = get(find(this.state.pageSearchAfter, {page: nextPage}), 'searchAfter', [])
+
+        //Calculate searchAfter of next page if not already calculated
+        if(isEmpty(searchAfter)) {
+            const lastCurrentTokenSort = get(last(this.state.tokens), 'sort', []);
+
+            const newEntry = {
+                page: nextPage,
+                searchAfter: lastCurrentTokenSort
+            }
+
+            this.setState({ pageSearchAfter: [...this.state.pageSearchAfter, newEntry]})
+            searchAfter = lastCurrentTokenSort
+        }
+
+        const tokens = await this.getTokens(searchAfter);
+        this.setState({tokens: tokens.hits, hasAfter: tokens.has_next, hasBefore: true, page: nextPage})
+    }
+
+    previousPageClicked = async(event) => {
+        const previousPage = this.state.page-1;
+        const searchAfter = get(find(this.state.pageSearchAfter, {page: previousPage}), 'searchAfter', [])
```

Comment:
What if we don't have this `searchAfter` in the state?

I understand that with "normal" use, you start on page 1, then to go back you have to have this element on state but we currently have a task to add the search params in the querystring so we can easily share links. In that case, you can start your component on page 2. In that case, how do you go back?

Reply:
We do have this limitation. We might need to limit querystring to `searchText`, `orderBy`, and `sortBy`

Reply:
I think it's fine for now and we can discuss that in the queryparam PR